### PR TITLE
Task review fixes

### PR DESCRIPTION
**Pull request by codymccoy**

### CODE REVIEW

File: View/Commands/SaveButtonEnabledConverter.cs [hunk 1]
```diff
@@ -21,19 +21,21 @@ class SaveButtonEnabledConverter : IMultiValueConverter
...
         {
-            bool hasErrors = (bool)values[0];
-            object dataContext = (values[1] as Button)?.DataContext;
+            //this is to answer the question Is Enabled. If HasFatalError is true or HasChanges is False,
+            //then we want to return false.
+            bool hasFatalError = (bool)values[0];
+            bool hasChanges = (bool)values[1];
 
-            if (hasErrors && dataContext is BaseViewModel baseVM)
+            if (hasFatalError == true)
```
Comment (RyanM-RMA) [inline View/Commands/SaveButtonEnabledConverter.cs:29]: This can be return !hasFatalError

File: ViewModel/BaseViewModel.cs [hunk 2]
```diff
@@ -11,46 +14,83 @@ namespace HEC.FDA.ViewModel
...
         #endregion
         #region Properties
 
-        public List<string> Errors
+        /// <summary>
+        /// The StudyCache holds all the elements used in FDA. You can use this to get any of them 
+        /// as well as listen for events where elements are added, removed, or updated
+        /// </summary>
+        public static IStudyCache StudyCache { get; set; }
+        public static Saving.PersistenceFactory PersistenceFactory { get; set; }
```
Comment (RyanM-RMA) [inline ViewModel/BaseViewModel.cs:42]: Namespacing

File: ViewModel/Watershed/TerrainOwnerElement.cs [hunk 3]
```diff
@@ -45,13 +46,22 @@ private void AddTerrainElement(object sender, ElementAddedEventArgs e)
...
-                .WithSiblingRules(this);
+            List<TerrainElement> terrainElems = StudyCache.GetChildElementsOfType<TerrainElement>();
+            if (terrainElems.Count == 0)
+            {
 
-            TerrainBrowserVM vm = new TerrainBrowserVM( actionManager);
-            string header = StringConstants.IMPORT_TERRAIN_HEADER;
-            DynamicTabVM tab = new DynamicTabVM(header, vm, StringConstants.IMPORT_TERRAIN_HEADER);
-            Navigate( tab, false,true);
+                Editors.EditorActionManager actionManager = new Editors.EditorActionManager()
```
Comment (RyanM-RMA) [inline ViewModel/Watershed/TerrainOwnerElement.cs:53]: Namespacing

**Review by RyanM-RMA — CHANGES_REQUESTED** [review]
Mostly some minor changes

File: Importer/AggregatedDamage/AggregateDamageFunction.cs [hunk 4]
```diff
@@ -54,9 +54,9 @@ public void SetSingleDamageFunction(StructureValueType typeValue, SingleDamageFu
         }
         public void Print(AsyncLogger logger)
         {
-            logger.Log($"\nAggregated Damage Function Name: {this.Name}");
-            logger.Log($"\tDescription: {this.Description}");
-            logger.Log($"\tCategory Name: {this.CategoryName}");
+            logger.Log($"\nAggregated Damage Function Name: ", this.Name);
```
[the remark this answers is not in the document]
Comment (codymccoy) [reply Importer/AggregatedDamage/AggregateDamageFunction.cs:57]: No. This is bob's code and he seems to like "this.". I did a find and replace and removed them all.

File: Importer/AggregatedDamage/AggregateDamageFunction.cs [hunk 5]
```diff
@@ -145,7 +145,7 @@ public void ExportData(StreamWriter wr, char delimt)
 
                 sdf = ObjectCopier.Clone(_SingleDamageFunction[ixType]);
 
-                //this.getSingleDamageFunction((StructureValueType)ixType, sdf);//Doesn't work
+                //getSingleDamageFunction((StructureValueType)ixType, sdf);//Doesn't work
```
Comment (RyanM-RMA) [inline Importer/AggregatedDamage/AggregateDamageFunction.cs:148]: Zombie

**Review by RyanM-RMA — CHANGES_REQUESTED** [review]
one last zombie

**Review by RyanM-RMA — APPROVED** [review]

**Pull request merged**